core/commands/store/challenge/challenge.go

```diff
@@ -125,34 +125,20 @@ the challenge request back to the caller.`,
 		if err != nil {
 			return err
 		}
-		// Check if host store has this contract id
-		shardInfo, err := storage.GetShardInfoFromDatastore(n, storage.HostStoragePrefix, sh)
-		if err != nil {
-			return err
-		}
-		ctID := req.Arguments[0]
-		if ctID != shardInfo.ContractID {
-			return fmt.Errorf("contract id does not match, has [%s], needs [%s]", shardInfo.ContractID, ctID)
-		}
-		if !shardHash.Equals(shardInfo.ShardHash) {
```

Comment:
hmm interesting. I think it's possible a shard hash maps to multiple contract ids ... better to remove this check like you did.